### PR TITLE
Feature: add command and parser support for addition of conditions

### LINKED ISSUE #24: Add a medical condition to a patient
As a private nurse I want to add a medical condition to a patient so that I can take note of the conditions they have

### LINKED ISSUE #180: Add parser and command support for add condition feature
Related to #24, #33 

No editing of conditions.

### PR DESCRIPTION
**Pull request by izzahaj**
Closes #24, closes #180  
This PR adds the ability to add a medical `Condition` to a `Patient`. Note that the conditions do not show up in the GUI, only in the `uninurse.jar` file.

For now, the syntax for adding a medical condition is `addCondition PATIENT_INDEX c/CONDITION`, since `AddGenericCommand` only supports addition of patients and tasks.

Most of the tests have been written for this feature, along with the `Condition` and `ConditionList` classes.

### CODE REVIEW

File: src/main/java/seedu/uninurse/logic/commands/AddConditionCommand.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Patient patientToEdit = lastShownList.get(index.getZeroBased());
+        ConditionList updatedConditionList = patientToEdit.getConditions().add(condition);
+        Patient editedPatient = new Patient(
+                patientToEdit.getName(), patientToEdit.getPhone(), patientToEdit.getEmail(),
+                patientToEdit.getAddress(), updatedConditionList, patientToEdit.getTasks(), patientToEdit.getTags());
```
Comment (BlopApple) [inline src/main/java/seedu/uninurse/logic/commands/AddConditionCommand.java:61]: I notice that the current way of modifying the Patient isn't very clean. Will create an issue to standardize a new method of creating a new immutable Patient without the long list of attributes.

File: src/test/java/seedu/uninurse/logic/commands/AddTaskCommandTest.java [hunk 2]
```diff
@@ -130,10 +130,10 @@ public void equals() {
         // null -> returns false
         assertNotEquals(null, addTaskFirstCommand);
 
-        // different person index -> returns false
+        // different task -> returns false
         assertNotEquals(addTaskFirstCommand, addTaskSecondCommand);
 
-        // different task index -> returns false
+        // different patient index -> returns false
```
Comment (BlopApple) [inline src/test/java/seedu/uninurse/logic/commands/AddTaskCommandTest.java:136]: I think "task" and "patient index" should be swapped. Should probably change the names of the `AddTaskCommand`s to be more fitting like the one in `AddConditionCommandTest`.

**Review by BlopApple — CHANGES_REQUESTED** [review]
Just a really small mistake that I notice when you tried to correct the comments in `AddTaskCommandTest`. Other than that LGTM.

File: src/test/java/seedu/uninurse/logic/commands/AddTaskCommandTest.java (hunk 2 above)
Comment (izzahaj) [reply src/test/java/seedu/uninurse/logic/commands/AddTaskCommandTest.java]: Thanks for pointing that out. I've updated the PR with your suggestions.

**Review by BlopApple — APPROVED** [review]
LGTM

**Pull request merged**